gdx/src/com/badlogic/gdx/maps/objects/TextMapObject.java

```diff
@@ -0,0 +1,269 @@
+/*******************************************************************************
+ * Copyright 2011 See AUTHORS file.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.maps.objects;
+
+import com.badlogic.gdx.maps.MapObject;
+import com.badlogic.gdx.utils.Align;
+
+/** @brief Represents a text map object */
+public class TextMapObject extends MapObject {
+
+	private float x = 0.0f;
+	private float y = 0.0f;
+	private float width = 0.0f;
+	private float height = 0.0f;
+	private float rotation = 0.0f;
+	private String text;
+	// Alignment values as represented via libGDX Align
+	private int horizontalAlign = Align.left;
+	private int verticalAlign = Align.top;
+	// String alignment values
+	private String tiledHorizontalAlign;
```

Comment:
I would just keep the String fields without the int fields because, as you pointed out, there is a  "justify" option for horizontal that LibGDX's `Align` does not have.

Reply:
So after I added in the option of also having the int values as well as the String. It was super convenient in the testing code being able to remove the giant alignment check before drawing it. I wonder if we should keep in on hand in case we ever add a justify?

Reply:
I don't have a strong opinion against it but the way I see it, users anyway have to do something with map objects because they are not rendered by the tiledmap renderer and most likely they have some game specific setup/parsing for map objects. 

E.g. in an ECS environment you will have an entity with a Text component or whatever, and you will then parse the TextObject and convert it to your own model. Doesn't matter if your parse the original strings or the int values. The string values though have the additional "justify" value which someone COULD do something with it.

But as mentioned, no problem if we keep it ;)

Reply:
Hmm... You are right. Maybe we should just try and follow the examples led by the other map objects. As far as I see there's no logic or utility methods in any of these. Circle just contains a Circle object, Polyline a Polyline and TextureMapObject contains a bunch of properties and a TextureRegion. No utility methods or logic. I guess maybe I'll go back to just storing the string values and not pollute it with all the extra stuff. Unless some other Tiledmap user chimes in and would like this. So I'll take my time with it.